deepsearch/cps/client/components/documents.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+from pathlib import Path
+import urllib3
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+from deepsearch.documents.core.convert import (
+    get_download_url,
+    download_converted_documents,
+)
+from deepsearch.documents.core.create_report import report_docs, report_urls
+
+if TYPE_CHECKING:
+    from deepsearch.cps.client import CpsApi
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+class DocumentConversionResult:
+    """
+    An instance of DocumentConversionResult is generated when document conversion is requested.
+    """
+
+    def __init__(
+        self,
+        proj_key: str,
+        task_ids: list,
+        statuses: List[str],
+        source_path: Optional[Path] = None,
+        source_urls: Optional[List[str]] = None,
+    ) -> None:
+        self.proj_key = proj_key
+        self.task_ids = task_ids
+        self.statuses = statuses
+        self._source_path = source_path
+        self._source_urls = source_urls
+
+    def download_json(self, result_dir: Path, progress_bar=False):
```

Comment:
Suggests this downloads _one_ JSON. But that is not what it looks like in the code. Please revise the use of plural and singular.